app/cardrecord.js

```diff
@@ -38,32 +38,25 @@ class CardRecorder extends MyTrello {
     const self = this;
     const deferred = Q.defer();
     const cardActions = self.cardFilters(card.actions, [{ name: 'comments', type: 'commentCard' }, { name: 'updates', type: 'updateCard' }, { name: 'created', type: 'createCard' }]);
-
-    self.deleteCurrentComment(cardActions.comments)
-      .then(() => {
-        const now = moment();
-        let hasMoved = false;
-        let daysSinceUpdate = false;
-        if (cardActions.updates.length > 0) {
-          hasMoved = DCH.hasMovedCheck(cardActions.updates);
-          daysSinceUpdate = now.diff(moment(cardActions.updates[0].date), 'days');
-        }
-        const totDays = (cardActions.comments.length > 0) ? DCH.calcTotalDays(cardActions.comments, now) : 0;
-        const prevMove = DCH.findPrevMoveDateFromComments({ commentList: cardActions.comments, actionList: cardActions.updates, createActionDate: cardActions.created.date });
-        self.inFinalList(card.idList)
-          .then(finalList => {
-            self.decideCommentType(card,
-              finalList,
-              daysSinceUpdate,
-              hasMoved,
-              prevMove,
-              cardActions,
-              totDays,
-              now)
-            .then(resp => { deferred.resolve(resp); })
-            .catch(deferred.reject);
-          });
-      });
+    const now = moment();
+    const recentlyMoved = self.checkRecentMove(cardActions.updates, now);
+    self.inFinalList(card.idList)
+    .then(finalList => {
+      if (finalList && !recentlyMoved) {
+        deferred.resolve({ 'final phase': true });
```

Comment:
This is inside the `then` callback so the deferred would have already been returned.  If you don't return here, the callback keeps executing and will call `self.deleteCurrentComment`?  Is it right for it to do that even if it's in the final phase already?